### ISSUE TITLE
Tests for cassandra-sink failing

### ISSUE THREAD

**Issue by styjdt** [spring-cloud/spring-cloud-stream-modules#225]
When running mvn clean install on spring-cloud-stream-modules the build fails due to test failures for `cassandra-sink`.

    Tests in error:
      CassandraSinkIntegrationTests$CassandraSinkIngestNamedParamsTests.org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraSinkIngestNamedParamsTests » NoHostAvailable
      CassandraSinkIntegrationTests$CassandraSinkIngestNamedParamsTests.testIngestQuery » IllegalState
      CassandraSinkIntegrationTests$CassandraSinkIngestTests.org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraSinkIngestTests » NoHostAvailable
      CassandraSinkIntegrationTests$CassandraSinkIngestTests.testIngestQuery » IllegalState

Each test ends with an exception which is eventually caused by 

    Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:52338 (com.datastax.driver.core.OperationTimedOutException: [localhost/127.0.0.1:52338] Operation timed out))

I guess, if the test only works with a running server instance it should be skipped dynamically, when no such server is running. As it is currently, there is no way to easily pull the repo and build it.

**Comment by ericbottard** [comment]
Alexander, to help you get going, you can use
```
mvn clean install -pl '!cassandra-sink'
```

We shall look into the underlying problem though (the tests use an embedded cassandra server, from what I could tell. Maybe your build environment is specific somehow. From other reports, seems you're on windows ?)

**Comment by styjdt** [comment]
My solution was just to comment out the cassandra-sink module in the parent pom.xml :smile: 

Yep, I'm on Windows 7, german version, also tried today on a VM with Windows Server 2012 R2 (english), with same result for this issue here...
But good to know, that Windows is considered as "specific somehow" :laughing:

**Comment by artembilan** [comment]
I'm also on Windows 7 x64, Intel i7 Quad.

Have just commented out `@DirtiesContext` on the `CassandraSinkIntegrationTests` and build is fine.

Looks like the problem is somewhere in the `cassandra-unit` and resources manipulations there.

Maybe we should consider to upgrade to some latest Cassandra Driver version and, therefore, test tool as well.

That may be some good kick to go ahead with `spring-integration-cassandra` development :smile: 

Try with the commented `@DirtiesContext` and with your confirmation we will push the fix into the `master`.

Thanks

**Comment by styjdt** [comment]
I'd like to test, but currently the build fails (without modification to yesterday) by

    [ERROR]   The project org.springframework.cloud.stream.module:spring-cloud-stream-modules:1.0.0.BUILD-SNAPSHOT (c:\Users\D053139\git\spring-cloud-stream-modules\pom.xml) has 1 error
    [ERROR]     'dependencies.dependency.version' for org.springframework.cloud:spring-cloud-stream-binder-redis:jar is missing. @ line 199, column 15

(The line numbers are probably little shifted w.r.t. your version, I've added a <proxies> section to get it working from corp network).
I do not understand enough from building against build-snapshots, but I guess that again there has been some change somewhere in the spring framework stack which now breaks this one?

**Comment by styjdt** [comment]
Okay, now I've fixed the version for `spring-cloud-stream-binder-redis` in all affected poms to `1.0.0.BUILD-20160317.060453-286`. I don't know it that is the correct way to solve such problems with snapshot versions, but at least I can reproduce the build of yesterday.

Unfortunately commenting out `@DirtiesContext` does not work fully. While yesterday I had more exception, today still one of the tests fail reproducibly:

    Running org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraEntityInsertTests
    Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 19.253 sec - in org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraEntityInsertTests
    Running org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraSinkIngestNamedParamsTests
    Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 47.395 sec - in org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraSinkIngestNamedParamsTests
    Running org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraSinkIngestTests
    Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 88.882 sec <<< FAILURE! - in org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraSinkIngestTests
    org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkIntegrationTests$CassandraSinkIngestTests  Time elapsed: 32.554 sec  <<< ERROR!
    com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:57356 (com.datastax.driver.core.OperationTimedOutException: [localhost/127.0.0.1:57356] Operation timed out))
        at com.datastax.driver.core.exceptions.NoHostAvailableException.copy(NoHostAvailableException.java:84)
        at com.datastax.driver.core.DriverThrowables.propagateCause(DriverThrowables.java:37)
        at com.datastax.driver.core.DefaultResultSetFuture.getUninterruptibly(DefaultResultSetFuture.java:214)
        at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:52)
        at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:36)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.dropKeyspacesWithNativeDriver(EmbeddedCassandraServerHelper.java:251)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.dropKeyspaces(EmbeddedCassandraServerHelper.java:205)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.cleanEmbeddedCassandra(EmbeddedCassandraServerHelper.java:162)
        at org.cassandraunit.spring.AbstractCassandraUnitTestExecutionListener.cleanServer(AbstractCassandraUnitTestExecutionListener.java:106)
        at org.cassandraunit.spring.CassandraUnitDependencyInjectionIntegrationTestExecutionListener.afterTestClass(CassandraUnitDependencyInjectionIntegrationTestExecutionListener.java:37)
        at org.springframework.test.context.TestContextManager.afterTestClass(TestContextManager.java:355)
        at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:77)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:193)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
    Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:57356 (com.datastax.driver.core.OperationTimedOutException: [localhost/127.0.0.1:57356] Operation timed out))
        at com.datastax.driver.core.RequestHandler.reportNoMoreHosts(RequestHandler.java:217)
        at com.datastax.driver.core.RequestHandler.access$1000(RequestHandler.java:44)
        at com.datastax.driver.core.RequestHandler$SpeculativeExecution.sendRequest(RequestHandler.java:276)
        at com.datastax.driver.core.RequestHandler$SpeculativeExecution$1.run(RequestHandler.java:374)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:812)

    Running org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkPropertiesTests
    Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.21 sec - in org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkPropertiesTests

Sorry, that is not a fix for me.

**Comment by mbogoevici** [comment]
@styjdt 

For the redis version issue, please update and rebase your changes to latest master.

**Comment by artembilan** [comment]
OK. There are several ways to go:
1. `@Ignore` those tests for Windows. If we confirm that they are stable with the Unix, of course...
2. Move them to the external resource with an appropriate `Rule`.
3. Upgrade to the latest Cassandra Driver and Unit for it with the hope that everything is going to work well. 

For the last one I'm afraid that we will must to modify Spring Integration Cassandra for those upgrades.

Any other ideas?

**Comment by styjdt** [comment]
@mbogoevici 
The redis issue is solved, thanks.

For the moment I've ignored all three tests for cassandra-sink. Although the insert test probably has some notion of dynamic skipping, because after running silent for long time (50 seconds up to 5 minutes) it ends in "skipped". So I've ignored that one too.
Since I don't use cassandra, but only wanted to report that the build is broken for Windows, I will not further follow up on that issue.

Thanks.
Alexander.
